test/acvp_client.py

```diff
@@ -189,12 +203,16 @@ def runTestSingle(promptName, prompt, expectedResultName, expectedResult, output
                 result = run_keyGen_test(tg, tc)
             tgResult["tests"].append(result)
 
+    # In case the testvectors are from the ACVTS server, it is expected
+    # that the acvVersion is included in the output results.
+    # See note on ACVTS data structure above.
+    if isAcvts is True:
+        results = [acvVersion, results]
+
     # Compare to expected results
     if expectedResult is not None:
         info(f"Comparing results with {expectedResultName}")
-        if json.dumps(results, sort_keys=True) != json.dumps(
-            expectedResult, sort_keys=True
-        ):
+        if json.dumps(results) != json.dumps(expectedResult):
```

Comment:
Hm, that seems a little dangerous? Does `json.dumps` make any guarantees about how a structure will be laid out?

Reply:
`json.dumps` uses `dict.iter` and since Python 3.7: `Dictionary order is guaranteed to be insertion order.`
https://docs.python.org/3.7/library/stdtypes.html#typesmapping
I'd like to strictly follow the order that NIST is using.

Reply:
👍 Can you add a comment referencing that guarantee?

Reply:
done.